File: backup/Sorting/python/14_카드정렬하기.py
# 이것이 취업을 위한 코딩테스트다
# https://www.acmicpc.net/problem/1715
# 1

import heapq

N = int(input())
arr = []
for _ in range(N):
    heapq.heappush(arr, int(input()))

result = 0
while len(arr) > 1:
    a = heapq.heappop(arr)
    b = heapq.heappop(arr)
    result += a + b
    heapq.heappush(arr, a + b)
print(result)

File: backup/Sorting/c++/14_카드정렬하기.cpp
// 이것이 취업을 위한 코딩테스트다
// https://www.acmicpc.net/problem/1715
// 1

#include <iostream>
#include <queue>

using namespace std;

int N;
priority_queue<int, vector<int>, greater<int>> pq;

int main() {
	cin >> N;
	for (int i = 0; i < N; i++) {
		int x;
		cin >> x;
		pq.push(x);
	}

	int result = 0;
	while (pq.size() > 1) {
		int a = pq.top(); pq.pop();
		int b = pq.top(); pq.pop();
		result += a + b;
		pq.push(a + b);
	}
	cout << result;
}